boxoffice/views/admin_report.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+
+from flask import Response, jsonify
+from .. import app, lastuser
+from coaster.views import load_models, render_with
+from boxoffice.models import db, ItemCollection, Order, Item, LineItem, Assignee, DiscountPolicy, LINE_ITEM_STATUS
+
+
+def jsonify_report(data_dict):
+    return jsonify(org_name=data_dict['item_collection'].organization.name, title=data_dict['item_collection'].title)
+
+
+@app.route('/admin/ic/<ic_id>/reports')
+@lastuser.requires_login
+@load_models(
+    (ItemCollection, {'id': 'ic_id'}, 'item_collection'),
+    permission='org_admin'
+    )
+@render_with({'text/html': 'index.html', 'application/json': jsonify_report}, json=True)
+def admin_report(item_collection):
+    return dict(title=item_collection.organization.title, item_collection=item_collection)
+
+
+def generate_csv(headers, rows):
+    for row in headers + rows:
+        yield ','.join([unicode(attr) for attr in row]) + '\n'
+
+
+@app.route('/admin/ic/<ic_id>/reports/tickets.csv')
+@lastuser.requires_login
+@load_models(
+    (ItemCollection, {'id': 'ic_id'}, 'item_collection'),
+    permission='org_admin'
+    )
+def admin_line_items_report(item_collection):
+    line_item_join = db.outerjoin(LineItem, Assignee).outerjoin(DiscountPolicy).join(Item).join(Order)
```

Comment:
This is because we also want to fetch line items that have not been assigned yet and line items that are not associated with any discounts.